vermouth/processors/merge_chains.py

```diff
@@ -42,8 +42,15 @@ def merge_chains(system, chains):
         The system to modify.
     chains: list[str]
         A container of chain identifier.
+    all_chains:
     """
-    chains = set(chains)
+    if not all_chains:
+        chains = set(chains)
+    else:
```

Comment:
What happens if both all_chains=True and chains is passed? IMO that should probably raise an error of some sorts.